torch/utils/data/dataloader.py

```diff
@@ -328,13 +342,24 @@ def __init__(self, loader):
         self._pin_memory = loader.pin_memory and torch.cuda.is_available()
         self._timeout = loader.timeout
         self._collate_fn = loader.collate_fn
-        self._sampler_iter = iter(self._index_sampler)
         self._base_seed = torch.empty((), dtype=torch.int64).random_().item()
-        self._num_yielded = 0
+        self._persistent_workers = loader._persistent_workers
+        self._reset_base()
```

Comment:
This is hacky. Can we eliminate `_reset_base` instead call `self._reset()` in `__iter__` before `return self`?

Reply:
Sure, let me give it a try :)